### ISSUE TITLE
Got all nan after training 4 epoch

### ISSUE THREAD

**Issue by mfxss** [chainer/chainer#2705]
I trained a [context2vec].(https://github.com/orenmel/context2vec/blob/master/context2vec/train/train_context2vec.py)
When using tensorflow, I can fix this by adding a small number to the loss function. 

> cross_entropy = self.target * tf.log(self.prediction+1e-10)

How to do this in chainer?

**Comment by delta2323** [comment]
What you want to do is to add a small number to the probability distribution that the prediction model outputs, isn't it?

**Comment by mfxss** [comment]
Since the built-in negative sampling cost function  should be stable.What else can I do to fix this?A slower learning rate or a different optimizer? Thank you for you advice.

**Comment by delta2323** [comment]
First of all, I recommend you where `nan` comes from. If we set `chainer.set_debug(True)`, Chainer raises an error as soon as `nan` occurs.

**Comment by mfxss** [comment]
If I want to set a lower learning rate, is the only way to change beta1 and bete2?
Also will a dropout help?
Thank you very much.

**Comment by delta2323** [comment]
I think `alpha` can be regarded as a learning rate in this particular case. Regarding dropout, I think we can get some insight whether dropout can help if we can understand the cause of `nan` more.
